### ISSUE TITLE
Authentication doesn't work for non LDAP users

### ISSUE THREAD

**Issue by dsolistorres** [dotCMS/core#16996]
Reported via Support: Ticket [98824](https://dotcms.zendesk.com/agent/tickets/98824)
Reproducible in dotCMS 5.1.6

Non LDAP users are not able to login when dotCMS is configured to work with LDAP for authentication.

### Steps to reproduce the behavior:
1. Configure dotCMS to work with LDAP for authentication (as described here: https://dotcms.com/docs/latest/ldap-configuration)
2. Create a user local to dotCMS without a matching LDAP user. The user cannot be assigned to the 'LDAP User' role in dotCMS. (users from starter site like 'admin@dotcms.com' can work if there are not LDAP users for them)
3. Try to authenticate using the above non LDAP user.
4. User is unable to login. The following stack trace is shown in the log:
```
ERROR enterprise.LDAPImpl: Error while trying to bind user admin@dotcms.com : 
com.dotcms.repackage.com.novell.ldap.LDAPReferralException: Referral 
at com.dotcms.repackage.com.novell.ldap.LDAPSearchResults.next(Unknown Source) ~[dot.ldap-ukv_2.jar:?] 
at com.dotcms.enterprise.LDAPImpl.a(SourceFile:189) ~[ee-5.1.6.jar:?] 
at com.dotcms.enterprise.LDAPImpl.authenticate(SourceFile:66) ~[ee-5.1.6.jar:?] 
at com.dotcms.enterprise.BaseAuthenticator.a(SourceFile:129) ~[ee-5.1.6.jar:?] 
at com.dotcms.enterprise.BaseAuthenticator.authenticateByEmailAddress(SourceFile:112) ~[ee-5.1.6.jar:?] 
at com.dotcms.enterprise.LDAPProxy.authenticateByEmailAddress(SourceFile:25) ~[ee-5.1.6.jar:?]
```

### Expected behavior
Native non-LDAP users should be able to login if these 2 conditions are met:
1. The username does not exist in LDAP
2. The user does not have the "LDAP User" role in dotCMS

### Additional context
This has been reported by customer via Support: Ticket [98824](https://dotcms.zendesk.com/agent/tickets/98824)

**Comment by nollymar** [comment]
PR: https://github.com/dotCMS/enterprise-2.x/pull/674/files

**Comment by nollymar** [comment]
**Note to QA:** Test cases that should be considered when LDAP Authentication is set: 

1. Log in with invalid LDAP and dotCMS user. It should fail
2. Log in with invalid LDAP and valid dotCMS user. It should log in  
3. Log in with valid LDAP. It should log in

**Comment by dsilvam** [comment]
Passed internal QA: Scenarios 1 and 2 tested. Scenario 3 will be tested on https://github.com/dotCMS/core/issues/17264

**Comment by bryanboza** [comment]
Fixed, tested on release-5.2 // Postgres // FF // openldap
